### PR TITLE
Improvements to Entity Outliner context menu

### PR DESCRIPTION
**Pull request by AMZN-mnaumov**
Removes "Create child entity", and instead creates entity based on current selection.
Instantiate Prefab instantiates prefab based on current selection.

### CODE REVIEW

**Review by AMZN-daimini — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp [hunk 1]
```diff
@@ -363,12 +364,25 @@ namespace AzToolsFramework
...
+                AZ::EntityId parentId;
+                AZ::Vector3 position = AZ::Vector3::CreateZero();
+
+                // if one entity is selected, set it as parent
+                EntityIdList selectedEntities;
+                ToolsApplicationRequestBus::BroadcastResult(selectedEntities, &ToolsApplicationRequests::GetSelectedEntities);
+                if (selectedEntities.size() == 1)
+                {
+                    parentId = selectedEntities.front();
+                    AZ::TransformBus::EventResult(position, parentId, &AZ::TransformInterface::GetWorldTranslation);
```
Comment (srikappa-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:376]: After this change, prefabs can be instantiated:
1. under a nested prefab of a level
2. under a child entity of level
3. under a child entity of nested prefab

Try to test that these 3 use cases work fine. Once these prefabs are instantiated, try to do more operations on them like modifying entities inside them, reparenting them, adding/removing entities under them, converting them into nested instances, etc.

Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:376]: Note that those actions were already possible before, but only when dragging and dropping a prefab file from the Asset Browser into the Outliner. So they were part of the testing when Instantiate was added.

We should definitely create automated tests though :)

**Comment by rainbj** [comment]
UX approved.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp (hunk 1 above)
Comment (AMZN-mnaumov) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:376]: @AMZN-daimini there are already tests for instantiating prefabs.

Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:376]: @AMZN-mnaumov only at the root level. The other cases srikappa mentioned should be added (over time).

**Review by srikappa-amzn — APPROVED** [review]

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 2]
```diff
@@ -646,16 +646,27 @@ void SandboxIntegrationManager::PopulateEditorGlobalContextMenu(QMenu* menu, con
...
+    }
+    // when a single entity is selected, entity is created as its child
+    else if (selected.size() == 1)
+    {
+        action = menu->addAction(QObject::tr("Create entity"));
+        QObject::connect(
+            action, &QAction::triggered, action,
+            [selected]
+            {
+                EBUS_EVENT(AzToolsFramework::EditorRequests::Bus, CreateNewEntityAsChild, selected.front());
```
Comment (gadams3) [inline Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:668]: Should expand this to get rid of EBUS_EVENT macro

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp [hunk 3]
```diff
@@ -646,16 +646,27 @@ void SandboxIntegrationManager::PopulateEditorGlobalContextMenu(QMenu* menu, con
...
+            action, &QAction::triggered, action,
+            [this]
+            {
+                ContextMenu_NewEntity();
+            });
+    }
+    // when a single entity is selected, entity is created as its child
+    else if (selected.size() == 1)
+    {
+        action = menu->addAction(QObject::tr("Create entity"));
```
Comment (gadams3) [inline Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:663]: Should this say "Create Child Entity"?

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp (hunk 2 above)
Comment (AMZN-mnaumov) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:668]: I was following the same pattern in that file, should I change that?

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp (hunk 3 above)
Comment (AMZN-daimini) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:663]: All these changes are making all context menu items contextual, so it shouldn't be necessary to change the labels to specify it.

Comment (AMZN-mnaumov) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:663]: no the UX wants to remove "Create child entity" and instead have "Create entity" handle both cases.

File: Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp (hunk 2 above)
Comment (AMZN-koppersr) [reply Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp:668]: Yeah generally expand the EBus macros, they should eventually be removed.

**Pull request merged**